language/complexity-check-plugin/src/complexity-check.ts

```diff
@@ -0,0 +1,328 @@
+      // if (
+      //   this.config.typeWeights &&
+      //   Object.keys(this.config.typeWeights).length > 0
+      // ) {
+      //   this.verboseDetails.push({
+      //     message: `----- Type Totals -----`,
+      //     severity: DiagnosticSeverity.Information,
+      //     range: diagnosticRange,
+      //   });
+      // }
+
+      // Object.entries(this.typeCount).forEach(([type, count]) => {
+      //   if (this.config.typeWeights) {
+      //     const typeMultiplier = this.config.typeWeights[type];
+      //     this.verboseDetails.push({
+      //       message: `${type}: ${count} x ${typeMultiplier} pt = ${
+      //         count * typeMultiplier
+      //       }`,
+      //       severity: DiagnosticSeverity.Information,
+      //       range: diagnosticRange,
+      //     });
+      //   }
+      // });
+
+      const message = `Content complexity is ${this.contentScore}`;
+
+      let diagnostic: Diagnostic = {
+        message: `${message}`,
+        severity: DiagnosticSeverity.Information,
+        range: diagnosticRange,
+      };
+
+      if (this.contentScore < this.config.maxAcceptableComplexity) {
+        if (
+          this.config.maxWarningLevel &&
+          this.contentScore > this.config.maxWarningLevel
+        ) {
+          diagnostic = {
+            message: `${message}, Warning: ${this.config.maxWarningLevel}, Maximum: ${this.config.maxAcceptableComplexity}`,
+            severity: DiagnosticSeverity.Warning,
+            range: diagnosticRange,
+          };
+        }
+      } else {
+        diagnostic = {
+          message: `${message}, Maximum: ${this.config.maxAcceptableComplexity}`,
+          severity: DiagnosticSeverity.Error,
+          range: diagnosticRange,
+        };
+      }
+
+      return [diagnostic, ...diagnostics, ...this.verboseDetails];
+    });
+  }
+
+  /** Create a validation visitor for dealing with transition states */
+  createContentChecker = (ctx: DocumentContext): ASTVisitor => {
+    return {
+      FlowNode: (flowNode) => {
+        this.range = {
+          start: flowNode.jsonNode.offset,
+          end: flowNode.jsonNode.offset + flowNode.jsonNode.length,
+        };
+      },
+
+      FlowStateNode: (flowState) => {
+        // add complexity per expression in a state
+        if (flowState.stateType) {
+          if (flowState.stateType.valueNode?.value === "ACTION") {
+            const numExp = getProperty(flowState, "exp");
+            if (numExp?.valueNode?.type === "array") {
+              this.contentScore += numExp.valueNode.children.length;
+
+              this.verboseDetails.push({
+                message: `state exp (+${numExp.valueNode.children.length}): ${this.contentScore}`,
+                severity: DiagnosticSeverity.Information,
+                range: toRange(ctx.document, flowState),
+              });
+            } else {
+              this.contentScore += 1;
```

Comment:
Should we definite all the static content scores we set somewhere as an object so that if/when we want to tweak these values we can do it in a singular place.